src/main/java/com/minecolonies/coremod/client/gui/townhall/WindowMainPage.java

```diff
@@ -64,111 +101,376 @@ public WindowMainPage(final BuildingTownHall.View building)
         registerButton(BUTTON_MERCENARY, this::mercenaryClicked);
         registerButton(BUTTON_TOWNHALLMAP, this::mapButtonClicked);
         registerButton(BUTTON_PATREON, this::patreonClicked);
+        registerButton(BUTTON_TOGGLE_JOB, this::toggleHiring);
+        registerButton(BUTTON_TOGGLE_HOUSING, this::toggleHousing);
+        registerButton(BUTTON_TOGGLE_MOVE_IN, this::toggleMoveIn);
+        registerButton(BUTTON_TOGGLE_PRINT_PROGRESS, this::togglePrintProgress);
+
+
+        registerButton(BUTTON_COLONY_SWITCH_STYLE, this::switchPack);
 
-        registerButton(BUTTON_TP, this::teleportToColony);
+        findPaneOfTypeByID(BUTTON_COLONY_SWITCH_STYLE, ButtonImage.class).setText(Component.literal(building.getColony().getStructurePack()));
+        registerButton(BUTTON_BANNER_PICKER, this::openBannerPicker);
+        registerButton(BUTTON_RESET_TEXTURE, this::resetTextureStyle);
+
+        this.colorDropDownList.setSelectedIndex(building.getColony().getTeamColonyColor().ordinal());
+        this.textureDropDownList.setSelectedIndex(TEXTURE_PACKS.indexOf(building.getColony().getTextureStyleId()));
+        this.initialTextureIndex = textureDropDownList.getSelectedIndex();
+
+        this.nameStyleDropDownList.setSelectedIndex(building.getColony().getNameFileIds().indexOf(building.getColony().getNameStyle()));
+        this.initialNamePackIndex = nameStyleDropDownList.getSelectedIndex();
+
+        checkFeatureUnlock();
+
+        if (building.getColony().isManualHiring())
+        {
+            findPaneOfTypeByID(BUTTON_TOGGLE_JOB, Button.class).setText(Component.translatable(COM_MINECOLONIES_COREMOD_GUI_HIRING_ON));
+        }
+
+        if (!building.getColony().isPrintingProgress())
+        {
+            findPaneOfTypeByID(BUTTON_TOGGLE_PRINT_PROGRESS, Button.class).setText(Component.translatable(OFF_STRING));
+        }
+
+        if (building.getColony().isManualHousing())
+        {
+            findPaneOfTypeByID(BUTTON_TOGGLE_HOUSING, Button.class).setText(Component.translatable(COM_MINECOLONIES_COREMOD_GUI_HIRING_ON));
+        }
+
+        if (building.getColony().canMoveIn())
+        {
+            findPaneOfTypeByID(BUTTON_TOGGLE_MOVE_IN, Button.class).setText(Component.translatable(ON_STRING));
+        }
     }
 
     /**
-     * On Patreon button clicked. Open website link to patreon.
+     * Toggles the allocation of a certain job. Manual or automatic.
+     *
+     * @param button the pressed button.
      */
-    private void patreonClicked()
+    private void toggleHiring(@NotNull final Button button)
     {
-        Minecraft.getInstance().setScreen(new ConfirmLinkScreen((check) -> {
-            if (check) {
-                Util.getPlatform().openUri("https://www.patreon.com/Minecolonies");
-            }
+        String key = button.getText().getContents() instanceof TranslatableContents ? ((TranslatableContents) button.getText().getContents()).getKey() : button.getTextAsString();
```

Comment:
Instanceof pattern matching
```suggestion
        String key = button.getText().getContents() instanceof final TranslatableContents contents ? contents.getKey() : button.getTextAsString();
```